server/queue/queue-api/src/main/java/org/apache/james/queue/api/MailQueue.java

```diff
@@ -95,7 +98,7 @@
      * Implementations should take care to do some kind of transactions to not
      * loose any mail on error
      */
-    MailQueueItem deQueue() throws MailQueueException, InterruptedException;
+    Flux<MailQueueItem> deQueue();

```

Comment:
This do mean that we are 100% OK to promote reactor as a TLD API.

Fair enough for me. I just want to check everybody agrees with this...

Reply:
I brought that discussion to mattermost last week (yes, you were in holiday, that's why I didn't want to go to fast on that topic).
I'm not comfortable with publishing Reactor types in our public APIs but I didn't found a better solution (In scala, we would use cats-effect to abstract the type but there's no such type in Java).
I have to look at reactive types that are part of the JDK to see what I can achieve.